protocol/icmpv6.go

```diff
@@ -0,0 +1,643 @@
+		n += i.Data.Len()
+	}
+	return 8 + i.Data.Len()
+}
+
+func (i *ICMPv6EchoReqRpl) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, int(i.Len()))
+	n := 0
+	b, err := i.ICMPv6Header.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	copy(data[n:], b)
+	n += len(b)
+	binary.BigEndian.PutUint16(data[n:], i.Identifier)
+	n += 2
+	binary.BigEndian.PutUint16(data[n:], i.SeqNum)
+	n += 2
+	dataBytes, err := i.Data.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	copy(data[n:], dataBytes)
+	return data, nil
+}
+
+func (i *ICMPv6EchoReqRpl) UnmarshalBinary(data []byte) error {
+	if len(data) < 4 {
+		return errors.New("The []byte is too short to unmarshal a full ICMPv6Header message.")
+	}
+	err := i.ICMPv6Header.UnmarshalBinary(data)
+	if err != nil {
+		return err
+	}
+	if len(data) < 8 {
+		return errors.New("The []byte is too short to unmarshal a full ICMPv6EchoReqRpl message.")
+	}
+	n := i.ICMPv6Header.Len()
+	i.Identifier = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	i.SeqNum = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	i.Data = new(util.Buffer)
+	err = i.Data.UnmarshalBinary(data[n:])
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewICMPv6EchoRequest(identifier, sequenceNumber uint16) *ICMPv6EchoReqRpl {
+	return &ICMPv6EchoReqRpl{
+		ICMPv6Header: ICMPv6Header{
+			Type: ICMPv6_Type_EchoRequest,
+			Code: 0,
+		},
+		Identifier: identifier,
+		SeqNum:     sequenceNumber,
+	}
+}
+
+func NewICMPv6EchoReply(identifier, sequenceNumber uint16) *ICMPv6EchoReqRpl {
+	return &ICMPv6EchoReqRpl{
+		ICMPv6Header: ICMPv6Header{
+			Type: ICMPv6_Type_EchoReply,
+			Code: 0,
+		},
+		Identifier: identifier,
+		SeqNum:     sequenceNumber,
+	}
+}
+
+type ICMPv6Error ICMPv6Header
+
+//     0                   1                   2                   3
+//    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   |     Type      |     Code      |          Checksum             |
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   |     Maximum Response Delay    |          Reserved             |
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//   |                                                               |
+//   +                                                               +
+//   |                                                               |
+//   +                       Multicast Address                       +
+//   |                                                               |
+//   +                                                               +
+//   |                                                               |
+//   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+type MLD struct {
+	ICMPv6Header
+	MaxResponse      uint16
+	Reserved         uint16
+	MulticastAddress net.IP
+}
+
+func (m *MLD) Len() (n uint16) {
+	return 24
+}
```

Comment:
Don't name the return variable 'n' if it is not being used.

func (m *MLD) Len() uint16 {

will be cleaner.

Reply:
yes, updated.